crates/ide_db/src/helpers.rs

```diff
@@ -282,3 +282,21 @@ pub fn for_each_break_expr(
         }
     }
 }
+
+/// Checks if the given lint is affected by the other lint which may or may not be a group.
+pub fn lint_is_affected(lint: &str, lint_is: &str) -> bool {
+    if lint == lint_is {
+        return true;
+    }
+
+    if let Some(group) = generated_lints::DEFAULT_LINTS
+        .iter()
+        .chain(generated_lints::CLIPPY_LINTS.iter())
+        .chain(generated_lints::RUSTDOC_LINTS.iter())
+        .find(|&check| check.label == lint_is)
+    {
```

Comment:
This could be quite slow given the sheer amount of lints there are, though it probably doesn't matter that much. Might make sense to revisit the lint stuff in general at some point(since there is also a specific fixme in completions)

Reply:
is there a way to cache it?

Reply:
Probably not, but I also missed your other comment about separating `Lint` and `LintGroup`. That sounds like a good idea to get around this here potentially.